aks.tf

```diff
@@ -0,0 +1,433 @@
+        max_surge                     = var.agents_pool_max_surge
+        drain_timeout_in_minutes      = var.agents_pool_drain_timeout_in_minutes
+        node_soak_duration_in_minutes = var.agents_pool_node_soak_duration_in_minutes
+      }
+    }
+  }
+
+  dynamic "aci_connector_linux" {
+    for_each = var.aci_connector_linux_enabled ? ["aci_connector_linux"] : []
+
+    content {
+      subnet_name = var.aci_connector_linux_subnet_name
+    }
+  }
+
+  dynamic "ingress_application_gateway" {
+    for_each = toset(var.ingress_application_gateway != null ? [var.ingress_application_gateway] : [])
+
+    content {
+      gateway_id   = ingress_application_gateway.value.gateway_id
+      gateway_name = ingress_application_gateway.value.gateway_name
+      subnet_cidr  = ingress_application_gateway.value.subnet_cidr
+      subnet_id    = ingress_application_gateway.value.subnet_id
+    }
+  }
+
+  dynamic "key_management_service" {
+    for_each = var.kms_enabled ? ["key_management_service"] : []
+
+    content {
+      key_vault_key_id         = var.kms_key_vault_key_id
+      key_vault_network_access = var.kms_key_vault_network_access
+    }
+  }
+
+  dynamic "key_vault_secrets_provider" {
+    for_each = var.key_vault_secrets_provider_enabled ? ["key_vault_secrets_provider"] : []
+
+    content {
+      secret_rotation_enabled  = var.secret_rotation_enabled
+      secret_rotation_interval = var.secret_rotation_interval
+    }
+  }
+
+  dynamic "kubelet_identity" {
+    for_each = var.kubelet_identity == null ? [] : [var.kubelet_identity]
+    content {
+      client_id                 = kubelet_identity.value.client_id
+      object_id                 = kubelet_identity.value.object_id
+      user_assigned_identity_id = kubelet_identity.value.user_assigned_identity_id
+    }
+  }
+
+  dynamic "http_proxy_config" {
+    for_each = var.enable_http_proxy ? [1] : []
+
+    content {
+      http_proxy  = var.http_proxy_config.http_proxy
+      https_proxy = var.http_proxy_config.https_proxy
+      no_proxy    = var.http_proxy_config.no_proxy
+    }
+  }
+
+
+  dynamic "http_proxy_config" {
+    for_each = var.http_proxy_config != null ? ["http_proxy_config"] : []
+
+    content {
+      http_proxy  = http_proxy_config.value.http_proxy
+      https_proxy = http_proxy_config.value.https_proxy
+      no_proxy    = http_proxy_config.value.no_proxy
+      trusted_ca  = http_proxy_config.value.trusted_ca
+    }
+  }
+
+  dynamic "confidential_computing" {
+    for_each = var.confidential_computing == null ? [] : [var.confidential_computing]
+
+    content {
+      sgx_quote_helper_enabled = confidential_computing.value.sgx_quote_helper_enabled
+    }
+  }
+
+  dynamic "confidential_computing" {
+    for_each = var.confidential_computing == null ? [] : [var.confidential_computing]
+
+    content {
+      sgx_quote_helper_enabled = confidential_computing.value.sgx_quote_helper_enabled
+    }
+  }
+
+  # dynamic "api_server_access_profile" {
```

Comment:
Why is this commented? If not required please remove it or uncomment.

Reply:
Uncommented